src/OrchardCore.Modules/OrchardCore.Roles/Permissions.cs

```diff
@@ -26,20 +26,14 @@ public async Task<IEnumerable<Permission>> GetPermissionsAsync()
     {
         var roleNames = (await _roleService.GetRoleNamesAsync())
             .Where(roleName => !RoleHelper.SystemRoleNames.Contains(roleName))
-            .ToList();
+            .ToArray();
 
-        var list = new List<Permission>(roleNames.Count + 3)
+        var list = new List<Permission>(roleNames.Length + 2)
```

Comment:
I don't understand why we need `roleNames` and a dynamically sized `List` in this method. Seems like a leftover that doesn't serve a purpose anymore.

Reply:
Yes, this must be removed. Dynamic permissions are not generated here anymore.

Reply:
agreed. I guess I was not paying attention.